### PR TITLE
ensure pennylane deprecation warnings are caught and removed

### PR DESCRIPTION
**Pull request by timmysilv**
**Context:**
Started by a discovery by @obliviateandsurrender that `qml.compile` raises a deprecation warning, because its `stop_at` uses `obj.name`. We don't need to validate the stopping condition for measurement processes if `expand_measurements` isn't set. Then, I thought that (in light of recent discussions about warnings being raised) maybe something is wrong with our pytest warnings situation. In fact, I accidentally started filtering out _all_ warnings when adding the PL deprecation warning type. So now I'm fixing all of those instances.

**Description of the Change:**
- remove `-p no:warnings` from pytest configuration and CI invocation
- Do not call `stop_at` on measurements in `expand_tape` if `expand_measurements` is False.
- fix/filter all failing tests
- update pennylane code as needed, kept as minimal as possible

**Benefits:**
- Avoid an unnecessary call to `stop_at` when we will force it to True for measurements anyway
- actually raise warnings as errors when we need to

**Possible Drawbacks:**
N/A

[sc-43949] [sc-55867]

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -204,6 +204,7 @@
   [(#5047)](https://github.com/PennyLaneAI/pennylane/pull/5047)
   [(#5071)](https://github.com/PennyLaneAI/pennylane/pull/5071)
   [(#5076)](https://github.com/PennyLaneAI/pennylane/pull/5076)
+  [()](https://github.com/PennyLaneAI/pennylane/pull/)
```
Comment (timmysilv) [inline doc/releases/changelog-dev.md:207]:
```suggestion
  [(#5122)](https://github.com/PennyLaneAI/pennylane/pull/5122)
```

**Review by albi3ro — APPROVED** [review]
Glad we tracked down what was silencing those warnings 😬

**Comment by timmysilv** [comment]
CI is now failing because of a new lightning change (fix in PennyLaneAI/pennylane-lightning#605) that refers to `mp.name`

**Review by lillian542 — APPROVED** [review]
Looks good to me 🎉

**Pull request merged**